src/ddscxx/include/dds/sub/detail/SampleRef.hpp

```diff
@@ -0,0 +1,118 @@
+#ifndef OMG_DDS_PUB_DETAIL_SAMPLEREF_HPP_
+#define OMG_DDS_PUB_DETAIL_SAMPLEREF_HPP_
+
+/* Copyright 2010, Object Management Group, Inc.
+ * Copyright 2010, PrismTech, Corp.
+ * Copyright 2010, Real-Time Innovations, Inc.
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+namespace dds
+{
+namespace sub
+{
+namespace detail
+{
+template <typename T>
+class SampleRef;
+}}}
+
+#include <dds/sub/SampleInfo.hpp>
+#include <org/eclipse/cyclonedds/topic/datatopic.hpp>
+//#include <dds/sub/detail/TSampleImpl.hpp>
```

Comment:
Might this be useful in the future, if not, remove?

Reply:
The file SampleRef.hpp was initially copied from Sample.hpp, and then modified to hold a Ref to the sample instead of a copy of the sample. The original Sample.hpp spec file also has this commented out include statement, but I agree that it would be better to remove it from this copy, and let the spec take care removing the original. Will correct this.